modular_zubbers/code/_onclick/hud/screen_objects/hud_timer.dm

```diff
@@ -0,0 +1,213 @@
+/// TODO: WEAKREFS ARE FUCKY WITH REMOVING
+/atom/movable/screen/text/screen_timer/proc/apply_to(list/mobs)
+	if(!islist(mobs))
+		if(!mobs)
+			return
+		mobs = list(mobs)
+	if(!length(timer_mobs) && length(mobs))
+		START_PROCESSING(SSprocessing, src)
+	for(var/player in mobs)
+		if(player in timer_mobs)
+			continue
+		if(istype(player, /datum/weakref))
+			var/datum/weakref/ref = player
+			player = ref.resolve()
+		attach(player)
+		RegisterSignal(player, COMSIG_MOB_LOGIN, PROC_REF(attach))
+		timer_mobs += WEAKREF(player)
+
+/// Removes the object from the client.screen of all mobs in the list, and unregisters the needed signals, while also stopping processing if there's no more mobs in the screen timers mob list
+/atom/movable/screen/text/screen_timer/proc/remove_from(list/mobs)
+	if(!islist(mobs))
+		if(!mobs)
+			return
+		mobs = list(mobs)
+	for(var/player in mobs)
+		UnregisterSignal(player, COMSIG_MOB_LOGIN)
+		if(player in timer_mobs)
+			timer_mobs -= player
+		if(istype(player, /datum/weakref))
+			var/datum/weakref/ref = player
+			player = ref.resolve()
+		de_attach(player)
+	if(!length(timer_mobs))
+		STOP_PROCESSING(SSprocessing, src)
+
+/// Updates the maptext to show the current time left on the timer
+/atom/movable/screen/text/screen_timer/proc/update_maptext()
+	var/time_formatted = time2text(timeleft(timer_id), "mm:ss")
+	var/timer_text = replacetextEx(maptext_string, "${timer}", time_formatted)
+	// If we don't find ${timer} in the string, just use the time formatted
+	var/result_text = "[maptext_style_left][timer_text][maptext_style_right]"
+	apply_change(result_text)
+
+/atom/movable/screen/text/screen_timer/proc/apply_change(result_text)
+	maptext = result_text
+
+/// Adds the object to the client.screen of the mob, or removes it if add_to_screen is FALSE
+/atom/movable/screen/text/screen_timer/proc/attach(mob/source, add_to_screen = TRUE)
+	SIGNAL_HANDLER
+	if(!source?.client)
+		return
+	var/client/client = source.client
+	// this checks if the screen is already added or removed
+	if(!can_attach(client, add_to_screen))
+		return
+	if(!ismob(source))
+		CRASH("Invalid source passed to screen_timer/attach()!")
+	do_attach(client, add_to_screen)
+
+/atom/movable/screen/text/screen_timer/proc/can_attach(client/client, add_to_screen)
+	return add_to_screen == (src in client.screen)
+
+/atom/movable/screen/text/screen_timer/proc/do_attach(client/client, add_to_screen)
+	if(add_to_screen)
+		client.screen += src
+	else
+		client.screen -= src
+
+/// Signal handler to run attach with specific args
+/atom/movable/screen/text/screen_timer/proc/de_attach(mob/source)
+	SIGNAL_HANDLER
+	attach(source, FALSE)
+
+/// Mainly a signal handler so we can run qdel()
+/atom/movable/screen/text/screen_timer/proc/delete_self()
+	SIGNAL_HANDLER
+	// I noticed in testing that even when qdel() is run, it still keeps running processing, even though it should have stopped processing due to running Destroy
+	STOP_PROCESSING(SSprocessing, src)
```

Comment:
This is redundant as this is already called in Destroy()

Reply:
This might be redundant on TGCode, TGMC where i took this from had the issue as specified in the comment, I'll test it